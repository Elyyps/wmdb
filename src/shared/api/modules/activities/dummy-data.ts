import { IActivitiesModule } from "./activities";
import INCENTIVE from "@assets/icons/incentive.svg";
import SPEL from "@assets/icons/spel.svg";
import SPORT from "@assets/icons/sportief.svg";
import ETEN from "@assets/icons/eten.svg";
import CREATIVE from "@assets/icons/creatief.svg";

import TEAMBUILDING from "@assets/icons/teambuilding.svg";
export const generateActivitiesDummyData = (): IActivitiesModule => ({
  items: [
    { color: "#b0cf6d", icon: ETEN, title: "Eten, Drinken & Feesten", total: 500 },
    { color: "#71abde", icon: TEAMBUILDING, title: "Teambuilden", total: 500 },
    { color: "#f8b272", icon: INCENTIVE, title: "Incentives", total: 500 },
    { color: "#ef7478", icon: SPEL, title: "Spel & Ontspanning", total: 500 },
    { color: "#d6b6d8", icon: SPORT, title: "Sportief & Actie(f)", total: 500 },
    { color: "#75c5c3", icon: CREATIVE, title: "Creatieve Uitjes", total: 500 },
    { color: "#ffdf7a", icon: INCENTIVE, title: "Workshops", total: 500 }
  ],
  title: "Bekijk ook andere activiteiten!"
});
