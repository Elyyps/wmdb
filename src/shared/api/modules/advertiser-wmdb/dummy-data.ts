import { IAdvertiserWMDB } from "./advertiser-wmdb";

export const generateAdvertiserDummyData = (): IAdvertiserWMDB => ({
  title: "Waarom bij ons adverteren",
  content: `<span>Door Wegmetdebaas.nl als marketingkanaal aan te spreken vergroot u de regio waarin uw klanten zich bevinden. </span>
  <br/>  <br/> 
  <ul className="uk-list-check">
<li>Groot bereik onder vele bedrijven</li>
<p>Via onze hoge posities bij Google en groot bereik via onze nieuwsbrief Gooves en via Facebook</p>
<li>Uitgebreide presentatie mogelijkheid</li>
<p>Uw teksten, foto’s en filmpjes perfect online gespresenteerd, ook op tablet en mobiel</p>
<li>Beheer zelf de inhoud van uw advertentie</li>
<p>Altijd mogelijk om zelf teksten en foto’s aan te passen</p>
<li>3 Maandelijkse rapportage</li>
<p>Uitgebreide rapportage van uw marketingactie(s)</p>
</ul> `,
  formTitle: "Graag ontvang ik meer informatie over adverteren op WegmetdeBaas.nl"
});
