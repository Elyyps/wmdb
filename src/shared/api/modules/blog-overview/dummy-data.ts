import { IBlogOverview } from "./blog-overview";
import ETEN from "@assets/icons/eten.svg";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";
const amountOfCards = 50;
export const generateBlogOverviewDummyData = (): IBlogOverview => ({
  title: "Alles over het kiezen van een locatie",
  filterOptions: [
    { title: "Vergaderen & Congres", icon: ETEN },
    { title: "Congres", icon: ETEN },
    { title: "Trouwen", icon: ETEN },
    { title: "Feest", icon: ETEN }
  ],
  cards: generateCards(amountOfCards)
});
export const generateCards = (amount: number): ICompanyOutingCard[] => {
  const totalCards: ICompanyOutingCard[] = [];
  for (let i = 0; i < amount; i += 1) {
    totalCards.push({
      title: `Dagje weg in de herfstvakantie? Bekijk deze musea eens! ${i}`,
      content:
        "Lorem ipsum dolor sit amet, consectetu dipiscing elit. Duis quis tortor lectus. Aliquam vel dignissim nunc",
      image: "https://picsum.photos/id/116/300/200",
      data: "18 sep. 2017",
      button: {
        title: "Lees meer",
        url: "/blog"
      }
    });
  }

  return totalCards;
};
