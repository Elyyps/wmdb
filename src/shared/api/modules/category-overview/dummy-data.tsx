import { ICategoryOverviewModule } from "./category-overview.model";
import SPEL from "@assets/icons/spel.svg";
import ETEN from "@assets/icons/eten.svg";
import SPORT from "@assets/icons/sportief.svg";
import CREATIVE from "@assets/icons/creatief.svg";
import INCENTIVE from "@assets/icons/incentive.svg";
import TEAM from "@assets/icons/teambuilding.svg";
const GenerateDummyCategoryOverviewData = (): ICategoryOverviewModule => ({
  backgroundColor: "#e8f2fc",
  paddingBottom: "72px",
  paddingTop: "72px",
  title: "Kies direct een categorie om het aanbod te bekijken!",
  items: [
    {
      title: "Eten, Drinken & Feesten",
      count: 500,
      image: "https://picsum.photos/id/1/600/400",
      icon: ETEN,
      link: "/single-wmdb"
    },
    {
      title: "Teambuilden",
      count: 430,
      image: "https://picsum.photos/id/20/600/400",
      icon: TEAM,
      link: "/single-wmdb"
    },
    {
      title: "Incentives",
      count: 430,
      image: "https://picsum.photos/id/30/600/400",
      icon: INCENTIVE,
      link: "/single-wmdb"
    },
    {
      title: "Spel & Ontspanning",
      count: 430,
      image: "https://picsum.photos/id/40/600/400",
      icon: SPEL,
      link: "/single-wmdb"
    },
    {
      title: "Sportief & Actie(f)",
      count: 430,
      image: "https://picsum.photos/id/50/600/400",
      icon: SPORT,
      link: "/single-wmdb"
    },
    {
      title: "Creative Uitjes",
      count: 20,
      image: "https://picsum.photos/id/60/600/400",
      icon: CREATIVE,
      link: "/single-wmdb"
    },
    {
      title: "Workshops",
      image: "https://picsum.photos/id/70/600/400",
      link: "/single-wmdb"
    },
    {
      title: "Helpdesk",
      image: "https://picsum.photos/id/80/600/400",
      link: "/single-wmdb"
    },
    {
      title: "Niewsbrief",
      image: "https://picsum.photos/id/90/600/400",
      link: "/single-wmdb"
    }
  ]
});

export { GenerateDummyCategoryOverviewData };
