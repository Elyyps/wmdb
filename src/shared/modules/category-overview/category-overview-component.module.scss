.category-overview {
  &__title {
    text-align: center;
    @media (min-width: $breakpoint-medium) {
      margin-bottom: 40px;
    }
  }

  &__posts {
    @media (max-width: $breakpoint-xsmall-max) {
      margin-left: -8px;
    }
    &:last-child {
      margin-bottom: -8px;
      @media (min-width: $breakpoint-small) {
        margin-bottom: -24px;
      }
      @media (min-width: $breakpoint-medium) {
        margin-bottom: -32px;
      }
    }

    > div {
      padding-bottom: 8px;
      @media (min-width: $breakpoint-small) {
        padding-bottom: 24px;
      }
      @media (min-width: $breakpoint-medium) {
        padding-bottom: 32px;
      }
      @media (max-width: $breakpoint-xsmall-max) {
        padding-left: 8px;
        &:last-child,
        &:nth-last-child(2) {
          width: 50%;
        }
      }
    }

    .category-overview__posts {
      margin-top: 0;
    }
  }
}
