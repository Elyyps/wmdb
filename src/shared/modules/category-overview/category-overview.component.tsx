import * as React from "react";
import styles from "./category-overview-component.module.scss";
import { CategoryCard } from "@app/core/category-card";
import {
  ICategoryOverviewModule,
  ICategoryCardModel
} from "@app/api/modules/category-overview/category-overview.model";
import { ModuleSectionComponent } from "@app/core/module-section";
import { withRouter, RouteComponentProps } from "react-router";

export interface ICategoryOverviewComponentProps {
  categoryOverviewModule: ICategoryOverviewModule;
}

const Component = (props: ICategoryOverviewComponentProps & RouteComponentProps) => {
  const { title, items, backgroundColor } = props.categoryOverviewModule;

  const openOverviewPage = (categoryCard: ICategoryCardModel) => {
    if (categoryCard.filterSectionId !== undefined) {
      let finalUrl = "/overzicht";

      // tslint:disable
      const foundSection = props.categoryOverviewModule.filterSections.find(
        section => section.id == categoryCard.filterSectionId
      );
      // tslint:enable
      if (foundSection) {
        const selectedItemsArrayString = foundSection.checkboxes.map(checkbox => checkbox.id).join("%2C");
        finalUrl = finalUrl.concat(`?categories=${selectedItemsArrayString}`);
      }

      props.history.push(finalUrl);
    } else {
      if (categoryCard.link) {
        props.history.push(categoryCard.link);
      }
    }
  };

  return (
    <ModuleSectionComponent backgroundColor={backgroundColor} paddingBottom="72px" paddingTop="72px">
      <div className={styles["category-overview"]}>
        <div className="uk-container">
          {title && <h2 className={styles["category-overview__title"]}>{title}</h2>}
          {items && (
            <div className={`${styles["category-overview__posts"]} uk-grid uk-child-width-1-2@s uk-child-width-1-3@m`}>
              {items.map((item, key) => (
                <div key={key}>
                  <CategoryCard
                    onClick={() => {
                      openOverviewPage(item);
                    }}
                    count={item.count}
                    icon={item.icon}
                    image={item.image}
                    link={item.link}
                    title={item.title}
                  />
                </div>
              ))}
            </div>
          )}
        </div>
      </div>
    </ModuleSectionComponent>
  );
};

const CategoryOverviewComponent = withRouter(Component);

export { CategoryOverviewComponent };
