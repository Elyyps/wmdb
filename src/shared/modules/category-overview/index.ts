export * from "./category-overview.component";
