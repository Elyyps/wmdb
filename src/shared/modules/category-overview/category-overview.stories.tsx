import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { CategoryOverviewComponent } from "./category-overview.component";
import { GenerateDummyCategoryOverviewData } from "@app/api/modules/category-overview/dummy-data";

storiesOf("CategoryOverview", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <CategoryOverviewComponent categoryOverviewModule={GenerateDummyCategoryOverviewData()} />
  ));
