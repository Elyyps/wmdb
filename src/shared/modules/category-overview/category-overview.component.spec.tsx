import React from "react";
import { shallow } from "enzyme";
import { CategoryOverviewComponent } from "./category-overview.component";
import { GenerateDummyCategoryOverviewData } from "@app/api/modules/category-overview/dummy-data";

describe("[CategoryOverview]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <CategoryOverviewComponent categoryOverviewModule={GenerateDummyCategoryOverviewData()} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
