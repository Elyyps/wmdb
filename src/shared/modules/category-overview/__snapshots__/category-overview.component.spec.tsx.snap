// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CategoryOverview] should render component without crashing 1`] = `
<ModuleSectionComponent
  backgroundColor="#e8f2fc"
  paddingBottom="72px"
  paddingTop="72px"
>
  <div
    className="category-overview"
  >
    <div
      className="uk-container"
    >
      <h2
        className="category-overview__title"
      >
        Kies direct een categorie om het aanbod te bekijken!
      </h2>
      <div
        className="category-overview__posts uk-grid uk-child-width-1-2@s uk-child-width-1-3@m"
      >
        <div
          key="0"
        >
          <CategoryCard
            count={500}
            icon="spel.svg"
            image="https://picsum.photos/id/1/600/400"
            link="/single-wmdb"
            title="Eten, Drinken & Feesten"
          />
        </div>
        <div
          key="1"
        >
          <CategoryCard
            count={430}
            icon="spel.svg"
            image="https://picsum.photos/id/20/600/400"
            link="/single-wmdb"
            title="Teambuilden"
          />
        </div>
        <div
          key="2"
        >
          <CategoryCard
            count={430}
            icon="spel.svg"
            image="https://picsum.photos/id/30/600/400"
            link="/single-wmdb"
            title="Incentives"
          />
        </div>
        <div
          key="3"
        >
          <CategoryCard
            count={430}
            icon="spel.svg"
            image="https://picsum.photos/id/40/600/400"
            link="/single-wmdb"
            title="Spel & Ontspanning"
          />
        </div>
        <div
          key="4"
        >
          <CategoryCard
            count={430}
            icon="spel.svg"
            image="https://picsum.photos/id/50/600/400"
            link="/single-wmdb"
            title="Sportief & Actie(f)"
          />
        </div>
        <div
          key="5"
        >
          <CategoryCard
            count={20}
            icon="spel.svg"
            image="https://picsum.photos/id/60/600/400"
            link="/single-wmdb"
            title="Creative Uitjes"
          />
        </div>
        <div
          key="6"
        >
          <CategoryCard
            image="https://picsum.photos/id/70/600/400"
            link="/single-wmdb"
            title="Workshops"
          />
        </div>
        <div
          key="7"
        >
          <CategoryCard
            image="https://picsum.photos/id/80/600/400"
            link="/single-wmdb"
            title="Helpdesk"
          />
        </div>
        <div
          key="8"
        >
          <CategoryCard
            image="https://picsum.photos/id/90/600/400"
            link="/single-wmdb"
            title="Niewsbrief"
          />
        </div>
      </div>
    </div>
  </div>
</ModuleSectionComponent>
`;
